app/models/teacher.rb

```diff
@@ -33,6 +36,8 @@ class Teacher < ApplicationRecord
   # TODO: remove after migration complete
   has_many :teacher_migration_failures
 
+  refresh_metadata -> { self }, on_event: %i[create update]
```

Comment:
will this cause issues with prod data as this is populated, I suppose there are no lead providers but worth confirming

Reply:
if there's no lead providers the call back won't update anything